js/src/common/states/AlertManagerState.ts

```diff
@@ -1,30 +1,41 @@
-import Alert from '../components/Alert';
+import Mithril from 'mithril';
+import Alert, { AlertAttrs } from '../components/Alert';
+
+export interface AlertState {
+  componentClass: AlertManagerState;
+  attrs: AlertAttrs;
+  children: Mithril.Children;
+}
 
 export default class AlertManagerState {
-  constructor() {
-    this.activeAlerts = {};
-    this.alertId = 0;
-  }
+  protected activeAlerts: { [ids: string]: AlertState } = {};
+  protected alertId = 0;
 
-  getActiveAlerts() {
+  public getActiveAlerts() {
     return this.activeAlerts;
   }
 
   /**
    * Show an Alert in the alerts area.
+   *
+   * @returns The alert's ID, which can be used to dismiss the alert.
    */
-  show(arg1, arg2, arg3) {
+  public show(children: Mithril.Children): number;
+  public show(attrs: AlertAttrs, children: Mithril.Children): number;
+  public show(componentClass: Alert, attrs: AlertAttrs, children: Mithril.Children): number;
```

Comment:
Could we return an explicit `AlertIdentifier` type here (that can extend from `number`, assuming this is possible in TS), to hide that implementation detail a little more? (And signal that this is an opaque value, not to be used except for memorizing the alert in question.)

Reply:
Done, good point